hack/e2e-test-ocp.sh

```diff
@@ -91,45 +95,77 @@ function collect_logs {
     $OC logs -n kueue-system -l app=kueue --tail=-1 > "$ARTIFACTS/kueue-system-logs.log" || true
     restore_ocp_manager_image
+    restore_kueue_namespace
 }
 
 function deploy_kueue {
-    # Update kueue image
-    (cd config/components/manager-ocp && $KUSTOMIZE edit set image controller="$IMAGE_TAG")
+    local namespace=${KUEUE_NAMESPACE:-kueue-system}
+    (cd config/components/manager-ocp && $KUSTOMIZE edit set image controller="$IMAGE_TAG" && \
+     $KUSTOMIZE edit set namespace "$namespace")
     
     # Deploy kueue
-    $OC apply --server-side -k config/default-ocp
+    local kustomize_path namespace
+    kustomize_path="config/default-ocp"
+
+    # Set namespace in the default-ocp Kustomization
+    (cd "${kustomize_path}" && \
+        $KUSTOMIZE edit set namespace "$namespace")
+
+    namespace="${KUEUE_NAMESPACE:-}"
+    
+    if [[ -n "$namespace" ]]; then
+        # Create a namespace if it doesn't exist
+        $OC create namespace "$namespace" --dry-run=client -o yaml | $OC apply -f -
+        
+        # Apply resources with namespace override
+        $KUSTOMIZE build "${kustomize_path}" | $OC apply --server-side -f -
+    else
+        # Apply directly without modifications
+        $OC apply --server-side -k "${kustomize_path}"
+    fi
 }
 
 function restore_ocp_manager_image {
     (cd config/components/manager-ocp && $KUSTOMIZE edit set image controller="$INITIAL_IMAGE")
 }
 
+function restore_kueue_namespace {
+    (cd config/default-ocp  && $KUSTOMIZE edit set namespace "$DEFAULT_NAMESPACE")
+}
+
 function deploy_cert_manager {
     echo "Deploying cert-manager..."
-    $OC apply -f config/default-ocp/cert_manager_rh.yaml
+    ${SOURCE_DIR}/deploy-cert-manager-ocp.sh
     wait_for_cert_manager_crds
     wait_for_cert_manager_csv
     wait_for_cert_manager_ready
 }
 
 trap collect_logs EXIT
-
-deploy_cert_manager
-deploy_kueue
+GINKGO_SKIP_PATTERN=""
+if [ "$SKIP_DEPLOY" != "true" ]; then
+    deploy_cert_manager
+    sleep 2m
+    deploy_kueue
+    # Skip e2e tests that depend on pod integration features,
+    # such as Deployment and StatefulSet, or other integrations not
+    # supported in OCP. Also, skip Alpha features like TAS.
+    # TODO: Remove Fair Sharing from the skip list once the issue is fixed.
+    GINKGO_SKIP_PATTERN="(AppWrapper|JobSet|LeaderWorkerSet|JAX|Kuberay|Metrics|Fair Sharing|TopologyAwareScheduling|Kueue visibility server|Failed Pod can be replaced in group|should allow to schedule a group of diverse pods|StatefulSet created with WorkloadPriorityClass)"
+else
+    echo "Skipping cert-manager and kueue deployment because SKIP_DEPLOY is set to true."
+    GINKGO_SKIP_PATTERN="(AppWrapper|JobSet|LeaderWorkerSet|JAX|Kuberay|Metrics|Fair Sharing|TopologyAwareScheduling|Kueue visibility server|Failed Pod can be replaced in group|should allow to schedule a group of diverse pods|StatefulSet created with WorkloadPriorityClass)"
```

Comment:
Is `StatefulSet created with WorkloadPriorityClass` a new test not present in release-0.11?

Reply:
Yea